pkg/envoy/cds/cluster.go

```diff
@@ -32,41 +33,16 @@ const (
 // replacer used to configure an Envoy cluster's altStatName
 var replacer = strings.NewReplacer(".", "_", ":", "_")
 
-type clusterOptions struct {
-	permissive             bool
-	withTLS                bool
-	withActiveHealthChecks bool
-}
-
-// clusterOption is type of function that edits the defaults of the options struct.
-type clusterOption func(o *clusterOptions)
-
-// withTLS is an option to disable TLS.
-func withTLS(o *clusterOptions) {
-	o.withTLS = true
-}
-
-// permissive is an option to not generate permissive endpoints discovery for clusters.
-func permissive(o *clusterOptions) {
-	o.permissive = true
-}
-
-// withActiveHealthChecks is an option to configure active health checks for upstream
-// clusters.
-func withActiveHealthChecks(o *clusterOptions) {
-	o.withActiveHealthChecks = true
-}
-
 // getUpstreamServiceCluster returns an Envoy Cluster corresponding to the given upstream service
 // Note: ServiceIdentity must be in the format "name.namespace" [https://github.com/openservicemesh/osm/issues/3188]
-// The defaults are non-permissive, and *no tls*.
-func getUpstreamServiceCluster(downstreamIdentity identity.ServiceIdentity, upstreamSvc service.MeshService, cfg configurator.Configurator, opts ...clusterOption) (*xds_cluster.Cluster, error) {
-	o := &clusterOptions{}
-	for _, opt := range opts {
-		opt(o)
```

Comment:
Let's retain the `ClusterOption` since it's an idiomatic way to avoid passing `cfg` and using mocks in test. You can keep that as is and remove only multicluster specific stuff.

Reply:
Added back the cluster options with only what is required by the current code